### ISSUE TITLE
Hello world is blank

### ISSUE THREAD

**Issue by fenhl** [maps4print/azul#174]
## Description

The hello world app (simple counter) produces a blank window.

## Version / OS

* azul version: 0.1.0 (8b7de76856cdcad5c448bc469238c61c3e9f046e)

* Operating system: Windows 10 Pro version 1803

* Windowing system (X11 or Wayland, Linux only):

## Steps to Reproduce

1. Recreate the [simple counter](https://github.com/maps4print/azul/wiki/A-simple-counter) example from the wiki, adjusting for the signature change to `Dom::with_callback`
2. Run the project
3. Observe that the window is blank

## Additional Information

 This appears to affect all Azul apps, not just this specific example.

It used to work previously, but reverting to a previous commit doesn't seem to fix it.

**Comment by fschutt** [comment]
Yes, I migrated the layout system to `stretch`, but I had to change things for performance reasons and there are currently no layout tests. I've fixed the wiki (for a start), but I can't observe that the window is blank, it looks like this:

![azul_hello_world](https://user-images.githubusercontent.com/12084016/59777068-33a91b80-92b4-11e9-8f05-d37a0c90fa29.png)

I currently don't have any windows computers around, but last time I looked it did look like this on windows, too. Since you say that all examples are like this, there's either something wrong with your graphics drivers or the screen isn't redrawing / updating.

**Comment by fenhl** [comment]
I've done some more testing and it turns out the blank screen bug only appears if the UI scale (Settings > System > Display > Change the size of text, apps, and other items) is above 100%. On 100% it looks mostly fine.

**Comment by Seong-su** [comment]
Same here.

![image](https://user-images.githubusercontent.com/48559964/61076114-747cf600-a456-11e9-9466-c6e708a19e3a.png)


## Additional Information


Found some strange symptoms.

1. Flickering the window during resizing it fastly.
2. With flickering, the button sometimes appears and disappears in a split second.


## Version / OS


* azul version: 32a3f11
* Operating system: Windows 10
* Rust version:
   * rustup 1.18.3 (435397f48 2019-05-22)
   * rustc 1.36.0 (a53f9df32 2019-07-03)
   * cargo 1.36.0 (c4fcfb725 2019-05-15)
* Rust toochain: stable-x86_64-pc-windows-msvc
* C compiler info: MSVC 2017

**Comment by SirTates** [comment]
I just tested this example too. If you resize the window the elements are redrawn and it looks as if the order of what is drawn is not constant a if the text is sometimes drawn before its parent, so it's obstructed.

**Comment by sci4me** [comment]
Yeah the entire framework is 100% broken on my machine :(

Azul version: 46e5f18f205c0ca0c66a9cfaa25f79929cd97c3d
OS: Arch Linux
Rust: 1.36.0

**Comment by 22116** [comment]
The same output on Solus 3 https://github.com/maps4print/azul/issues/174#issuecomment-510605139

**Comment by jamesrobb** [comment]
I'm also experiencing the same problem as @Seong-su 

The counter example is also broken. The labels sometimes appear if I resize the window, but quickly disappear again.

Azul version: 5984b8020ca7bbf218af1704243f277f34da600b
OS: Arch Linux
Rust: 1.37.0

**Comment by foxboxpdx** [comment]
Said I'd post updates in this ticket so here's a positive update:

Compiling on Win10, Cargo 1.38, branch=unvendor_dependencies#c7adea0d

After tweaking my main() based on the changes seen in app.rs (sending the window options as an argument to run() instead of .create_window()), I was able to successfully populate a window with widgets and data.  This is with UI scale above 100% and all that jazz.  It doesn't necessarily look good, mind you, but at least it's displaying stuff now.

**Comment by jhwgh1968** [comment]
OS: Arch Linux
Rust version: rustc 1.40.0-nightly (518deda77 2019-10-18)

The screen shot @fschutt took of the example app looks correct on my machine, but I also see the bad calculator screen shot @Seong-su posted.

When I tried the branch @foxboxpdx suggested (which has been updated to 04489ea761da211c621bf51a8b1336608eb1e516), I get a consistently blank screen for the Hello World app and the Calculator looks worse.

Funny addendum: when I run the calculator example, I get _four copies_ of the window for some reason. On master, closing any of the windows closes all four.

I think something very strange is going on. What platforms do the layout tests run on? I wonder if Arch Linux has a different package version than the Ubuntu/Debian-based distros that is causing us extra trouble.

**Comment by fschutt** [comment]
@jhwgh1968 The old layout solver had no flexbox tests, and the new layout solver isn't yet finished (hence the blank screen). Just wait. This repository isn't ready yet.
